app/utils/build/__init__.py

```diff
@@ -230,106 +237,99 @@ def parse_build_data(build_data, job, kernel, errors, build_dir=None):
-            build_doc.git_describe = data_pop(
-                models.GIT_DESCRIBE_KEY, None)
-            build_doc.git_url = data_pop(models.GIT_URL_KEY, None)
-            build_doc.kconfig_fragments = kconfig_fragments
-            build_doc.kernel_config = data_pop(
-                models.KERNEL_CONFIG_KEY, None)
-            build_doc.kernel_image = data_pop(
-                models.KERNEL_IMAGE_KEY, None)
-            build_doc.modules = data_pop(models.MODULES_KEY, None)
-            build_doc.modules_dir = data_pop(
-                models.MODULES_DIR_KEY, None)
-            build_doc.status = data_pop(
-                models.BUILD_RESULT_KEY, models.UNKNOWN_STATUS)
-            build_doc.system_map = data_pop(
-                models.SYSTEM_MAP_KEY, None)
-            build_doc.text_offset = data_pop(
-                models.TEXT_OFFSET_KEY, None)
-            build_doc.version = data_pop(models.VERSION_KEY, "1.0")
-            build_doc.warnings = data_pop(models.BUILD_WARNINGS_KEY, 0)
-            build_doc.kernel_image_size = data_pop(
-                models.KERNEL_IMAGE_SIZE_KEY, None)
-            build_doc.modules_size = data_pop(models.MODULES_SIZE_KEY, None)
-            build_doc.cross_compile = data_pop(models.CROSS_COMPILE_KEY, None)
-
-            build_doc.git_describe_v = data_pop(
-                models.GIT_DESCRIBE_V_KEY, None)
-            build_doc.kernel_version = _extract_kernel_version(
-                build_doc.git_describe_v, build_doc.git_describe)
-
-            compiler_version_full = (
-                data_pop(models.COMPILER_VERSION_FULL_KEY, None) or
-                data_pop(models.COMPILER_VERSION_KEY, None))
-
-            compiler_data = _extract_compiler_data(compiler_version_full)
-            build_doc.compiler = compiler_data[0]
-            build_doc.compiler_version = compiler_data[1]
-            build_doc.compiler_version_ext = compiler_data[2]
-            build_doc.compiler_version_full = compiler_data[3]
-
-            artifacts = {
-                models.BUILD_LOG_SIZE_KEY: build_doc.build_log,
-                models.KERNEL_CONFIG_SIZE_KEY: build_doc.kernel_config,
-                models.KERNEL_IMAGE_SIZE_KEY: build_doc.kernel_image,
-                models.MODULES_SIZE_KEY: build_doc.modules,
-                models.SYSTEM_MAP_SIZE_KEY: build_doc.system_map,
-                models.VMLINUX_FILE_SIZE_KEY:
-                    data_pop(models.VMLINUX_FILE_KEY, None)
-            }
-
-            build_doc.metadata = build_data
-        except KeyError, ex:
-            err_msg = (
-                "Missing mandatory key in build data (job: %s, kernel: %s)")
-            utils.LOG.exception(ex)
-            utils.LOG.error(err_msg, job, kernel)
-            ERR_ADD(errors, 500, err_msg % (job, kernel))
-    else:
-        utils.LOG.error("Provided data does not look like json")
-        ERR_ADD(errors, 500, "Provided data is not json")
+        build_doc.metadata = build_data
```

Comment:
pop-ing works better for selecting which values were used, not values validated against. Nevertheless, a copy of the input data should be used if `pop` is the right thing to do.